src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -168,14 +168,18 @@ impl <'a, 'b> ReadOnlyChecker <'a, 'b> {
                 self.check_all_read_only(args)
             },
             AtBlock => {
+                check_argument_count(2, args)?;
```

Comment:
We're sure that `check_argument_count()` is getting called for each type variant now, right?  Not just in the read-only checker, but in all analysis passes where it matters?

Reply:
Yep -- the only analysis pass which occurs _before_ the type checker (which calls check_argument_count for all function types) is the read-only checker. The only AST pass that tries to directly read arguments also checks before doing so.